src/components/SelectionListWithModal/index.tsx

```diff
@@ -23,13 +23,15 @@ function SelectionListWithModal<TItem extends ListItem>(
     const [isModalVisible, setIsModalVisible] = useState(false);
     const [longPressedItem, setLongPressedItem] = useState<TItem | null>(null);
     const {translate} = useLocalize();
-    const {shouldUseNarrowLayout} = useResponsiveLayout();
+    // We need to use isSmallScreenWidth instead of shouldUseNarrowLayout https://github.com/Expensify/App/issues/48675
+    // Because there is a race condition somewhere in the app that causes shouldUseNarrowLayout to change indefinitely in this component
```

Comment:
How bout we make this comment like this? And please copy this into `src/pages/workspace/reportFields/ReportFieldsListValuesPage.tsx` as well

```suggestion
    // We need to use isSmallScreenWidth instead of shouldUseNarrowLayout here because there is a race condition that causes shouldUseNarrowLayout to change indefinitely in this component
    // See https://github.com/Expensify/App/issues/48675 for more details
```